www/src/components/docs-table-of-contents.js

```diff
@@ -14,54 +14,51 @@ function isUnderDepthLimit(depth, maxDepth) {
 // max depth is set via the tableOfContentsDepth field in the frontmatter, all headings will be rendered
-function createItems(items, location, depth, maxDepth) {
+function createItems(items, location, depth, maxDepth, activeHash) {
   return (
     items &&
-    items.map((item, index) => (
-      <li
-        sx={{ [mediaQueries.xl]: { fontSize: 1 } }}
-        key={location.pathname + (item.url || depth + `-` + index)}
-      >
-        {item.url && (
-          <Link
-            sx={{
-              "&&": {
-                color: `textMuted`,
-                border: 0,
-                transition: t =>
-                  `all ${t.transition.speed.fast} ${t.transition.curve.default}`,
-                ":hover": {
-                  color: `link.color`,
-                  borderBottom: t => `1px solid ${t.colors.link.hoverBorder}`,
+    items.map((item, index) => {
+      const isActive = item.url === `#${activeHash}`
+      return (
+        <li
+          sx={{ [mediaQueries.xl]: { fontSize: 1 } }}
+          key={location.pathname + (item.url || depth + `-` + index)}
+        >
+          {item.url && (
+            <Link
+              sx={{
+                "&&": {
+                  color: isActive ? `link.color` : `textMuted`,
+                  border: 0,
+                  borderBottom: t =>
+                    isActive
+                      ? `1px solid ${t.colors.link.hoverBorder}`
+                      : `none`,
+                  transition: t =>
+                    `all ${t.transition.speed.fast} ${t.transition.curve.default}`,
+                  ":hover": {
+                    color: `link.color`,
+                    borderBottom: t => `1px solid ${t.colors.link.hoverBorder}`,
+                  },
                 },
-              },
-            }}
-            getProps={({ href, location }) =>
-              location && location.href && location.href.includes(href)
-                ? {
-                    style: {
-                      color: colors.link.color,
-                      borderBottom: `1px solid ${colors.link.hoverBorder}`,
-                    },
-                  }
-                : null
-            }
-            to={location.pathname + item.url}
-          >
-            {item.title}
-          </Link>
-        )}
-        {item.items && isUnderDepthLimit(depth, maxDepth) && (
-          <ul sx={{ color: `textMuted`, listStyle: `none`, ml: 5 }}>
-            {createItems(item.items, location, depth + 1, maxDepth)}
-          </ul>
-        )}
-      </li>
-    ))
+              }}
+              to={location.pathname + item.url}
+            >
+              {item.title}
+            </Link>
+          )}
+          {item.items && isUnderDepthLimit(depth, maxDepth) && (
+            <ul sx={{ color: `textMuted`, listStyle: `none`, ml: 5 }}>
+              {createItems(item.items, location, depth + 1, maxDepth)}
```

Comment:
```suggestion
              {createItems(item.items, location, depth + 1, maxDepth, activeHash)}
```

is `activeHash` missing here in case subitems are also selected?

Reply:
Yes, I forgot to add the argument. Fixed.